morpheus/_lib/doca/src/doca_source.cpp

```diff
@@ -214,139 +149,71 @@ DocaSourceStage::subscriber_fn_t DocaSourceStage::build()
-
-                // Do we still need this synchronize?
-                //  const auto gather_meta_stop = now_ns();
-
-                cudaStreamSynchronize(pstream_cpp);
-                output.on_next(std::move(meta));
-
-                m_semaphore[thread_idx]->set_free(sem_idx);
-                sem_idx = (sem_idx + 1) % MAX_SEM_X_QUEUE;
-
-                // const auto end = now_ns();
-                // LOG(WARNING) << "Queue " << thread_idx
-                //             << " packets " << packet_count
-                //             << " kernel time ns " << start - start_kernel
-                //             << " CPU time ns " << end - start
-                //             << " table creation ns " << table_stop - start
-                //             << " gather payload ns " << gather_payload_stop - table_stop
-                //             << " table create ns " << table_create_stop - gather_payload_stop
-                //             << " gather column ns " << gathered_columns_stop - table_create_stop
-                //             << " gather meta schema ns " << gather_table_meta - gathered_columns_stop
-                //             << " gather meta table ns " << create_message_cpp - gather_table_meta
-                //             << " create message cpp ns " << gather_meta_stop - create_message_cpp
-                //             << " final ns " << end - gather_meta_stop
-                //             << std::endl;
+                    // LOG(WARNING) << "pkts " << pkt_ptr->packet_count_out << " MAX_PKT_SIZE " << MAX_PKT_SIZE;
+                    // std::endl;
+
+                    auto meta = RawPacketMessage::create_from_cpp(pkt_ptr->packet_count_out,
+                                                                  MAX_PKT_SIZE,
+                                                                  pkt_ptr->pkt_addr,
+                                                                  pkt_ptr->pkt_hdr_size,
+                                                                  pkt_ptr->pkt_pld_size,
+                                                                  true,
+                                                                  queue_idx);
+
+                    // const auto create_msg = now_ns();
+
+                    output.on_next(std::move(meta));
+
+                    m_semaphore[queue_idx]->set_free(sem_idx[queue_idx]);
+                    sem_idx[queue_idx] = (sem_idx[queue_idx] + 1) % MAX_SEM_X_QUEUE;
+
+                    // const auto end = now_ns();
+
+                    // LOG(WARNING) << "Queue " << queue_idx
+                    //             << " packets " << pkt_ptr->packet_count_out
+                    //             << " kernel time ns " << end_kernel - start_kernel
+                    //             << " Sem + msg ns " << create_msg - start_sem
+                    //             << " End ns " << end - create_msg
+                    //             << std::endl;
+                }
```

Comment:
There is a lot of commented out timing code. You can use a `#define` to remove these from the compilation or just delete the comments